language/move-lang/tests/functional/shared_ed25519_public_key/shared_key.move

```diff
@@ -61,3 +64,19 @@ fun main() {
 }
 // check: ABORTED
 // check: 7000
+
+// rotating to a key with a good length but bad contents should fail
```

Comment:
Do we want to have separate tests for failure cases:
1) point not on curve,
2) point of small order,
3) point not canonically serialized?

Reply:
Not for now, as the native function does not expose them.
 It may be an improvement if Move has the error code breadth.

Reply:
I think it would be useful to have these tests + maybe a few others (e.g. key too long, key too short) in `functional/natives/signature.move`. I don't think we want/need a more sophisticated error codes to write these tests--checking that the function says `false` for all of them is enough.

Reply:
Done